### PR TITLE
camera: add set_mode (synchronous)

### PR DESCRIPTION
**Pull request by JonasVautherin**
Add a synchronous `set_mode` function to the camera interface.

Connects to #334.

### CODE REVIEW

**Comment by JonasVautherin** [comment]
I will add integration tests in `integration_tests/camera_mode.cpp`.

File: plugins/camera/camera_impl.cpp [hunk 1]
```diff
@@ -461,6 +483,7 @@ void CameraImpl::get_mode_async(Camera::mode_callback_t callback)
     if (_get_mode.callback != nullptr) {
         if (callback) {
             callback(Camera::Result::BUSY, Camera::Mode::UNKNOWN);
+            return;
```
Comment (julianoes) [inline plugins/camera/camera_impl.cpp]: Nice catch!

File: plugins/camera/camera_impl.cpp [hunk 2]
```diff
@@ -430,23 +431,44 @@ Camera::Result CameraImpl::camera_result_from_command_result(MAVLinkCommands::Re
...
+
+void CameraImpl::save_camera_mode(const float mavlink_camera_mode)
+{
+    MAVLinkParameters::ParamValue value;
+    value.set_uint32(mavlink_camera_mode);
+    _camera_definition->set_setting("CAM_MODE", value);
+    refresh_params();
+}
+
+float CameraImpl::to_mavlink_camera_mode(const Camera::Mode mode) const
```
Comment (julianoes) [inline plugins/camera/camera_impl.cpp]: :+1:

File: plugins/camera/camera_impl.cpp [hunk 3]
```diff
@@ -627,25 +650,11 @@ void CameraImpl::process_camera_settings(const mavlink_message_t &message)
...
-            mode = Camera::Mode::PHOTO;
-            break;
-        case CAMERA_MODE_VIDEO:
-            mode = Camera::Mode::VIDEO;
-            break;
-        default:
-            mode = Camera::Mode::UNKNOWN;
-            break;
-    }
+    const auto mode = to_camera_mode(camera_settings.mode_id);
```
Comment (julianoes) [inline plugins/camera/camera_impl.cpp]: I like the improvements.

**Pull request merged**